### ISSUE TITLE
ActionButton label doesn't use disabled text color in high contrast mode

### ISSUE THREAD

**Issue by molinger** [microsoft/fluentui#16125]
Possibly related to #9418.

### Environment Information

- **Package version(s)**: 7.153.3+8.0.0-beta.23
- **Browser and OS versions**: Chromium Edge + Windows 10

### Describe the issue:

1. Enable high contrast mode in Windows.
2. https://fabricweb.z5.web.core.windows.net/oufr/7.153.3/#/examples/button
   - Isolated repro: https://codepen.io/justslone-the-encoder/pen/poEyPZV?editors=1010
3. Disable buttons using toggle.

Image:
![image](https://user-images.githubusercontent.com/40304143/101261699-0f324080-36ee-11eb-8e45-d00011e2ebad.png)

#### Actual behavior:

Action button label uses text color.

#### Expected behavior:

Action button label uses disabled text color like other buttons do. Note that the primary button in the split button examples also uses the text color when disabled. Every other button uses the disabled text color though for the label.

**Comment by JustSlone** [comment]
@molinger thanks for the issue! Confirmed the repro, updated the issue with a isolated CodePen.

Looks like the issue is here: the disabled color for action button is specified here but there is no high contrast selector.
https://github.com/microsoft/fluentui/blob/9c92b506d8139d1ab596caaec276774d9ae27b81/packages/react-internal/src/compat/components/Button/ActionButton/ActionButton.styles.ts#L47-L51

See this line for the default button styles:
https://github.com/microsoft/fluentui/blob/9c92b506d8139d1ab596caaec276774d9ae27b81/packages/react-internal/src/compat/components/Button/ButtonThemes.ts#L64-L72

It seems like adding a HighContrastSelector and just setting  color: 'GrayText', should be sufficient to fix this issue. 

@molinger are you interested in spinning up a PR and giving a go at fixing this issue?

**Comment by molinger** [comment]
I wrote up the suggested fix and it works on Edge. I'll do a little more testing and send out a PR.
